CHI_TECH/ChiMath/Quadratures/quadrature_gausslegendre.cc

```diff
@@ -12,64 +20,125 @@ void chi_math::QuadratureGaussLegendre::
   {
     printf("Initializing Gauss-Legendre Quadrature with %d q-points\n",N);
   }
-  //============================================= InitializeAlphaElements init guess and weights
-  double dx = 2.0/2000;
-  double xgold = -1.0;
-  double fold = Legendre(N,xgold);
-  //printf("Fold=%f\n",fold);
-  double xgnew = 0.0;
-  double fnew = 0.0;
-  for (int i=0;i<2000;i++)
+
+  //========================= Compute the roots
+  FindRoots(N, abscissae, maxiters, tol);
+
+  //========================= Compute the weights
+  weights.resize(N,1.0);
+  for (size_t k=0; k < abscissae.size(); k++)
   {
-    xgnew = xgold + dx;
-    fnew = Legendre(N,xgnew);
-    //printf("xgnew=%f Fnew=%f  Fold=%f\n",xgnew,fnew,fold);
-    if ((fnew*fold)<0.0)
-    {
-      abscissae.push_back(xgnew);
-      weights.push_back(1.0);
-    }
+    weights[k] =
```

Comment:
IMO, `f = a/(b*c*d*e)` is more understandable than `f = a/b/c/d/e` and it could also be a tiny bit faster (multiplication should be generally faster than division, see e.g. [here](https://stackoverflow.com/questions/4125033/floating-point-division-vs-floating-point-multiplication/5322101)). I suggest rewriting using multiplication, but if other reviewers are fine with that and you want to keep the divisions, I won't fight for it :-) 

As a side note - I became curious, and since I was reading this right after today's 689 class where I was doing some benchmarking, I wrote a small quick [test](http://quick-bench.com/ZvhKG5wQsG34fKt1V0T-jh7gzGM)... you'll see that the performance is probably not what we should be concerned about here, but for me it is really the readability...